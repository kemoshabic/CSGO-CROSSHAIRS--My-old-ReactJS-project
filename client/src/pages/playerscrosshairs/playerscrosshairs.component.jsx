import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectFavoriteCrosshairs } from "../../redux/settings/settings.selector";
import { selectIfCopied } from "../../redux/settings/settings.selector";
import { removeCrosshair } from "../../redux/settings/settings.actions"

import Copy from '../../components/Copy/copy.component'
import PinkHeart from "../../assets/pink.png";

import "./playerscrosshairs.styles.scss";

export const PlayersCrosshairs = ({ savedCrosshairs, copied, removeCrosshair }) => {
  return (
    <div className="crosshair">
    <div className={`${copied && 'info__copied-succes'} info__copied`}> <p className="info__copied-text">{copied && 'Copied to clipboard'}</p></div>
      
    {savedCrosshairs.map(({ crosshair, id, name, team, imageUrl}) => (
        <div className="crosshair__wrap" key={id}>
          <img src={PinkHeart} alt="pink heart" />
          <h2 className="crosshair__heading">{name}</h2>
          <span className="crosshair__team">{team}</span>
          <Copy crosshair={crosshair}/>
          <div className="crosshair__img" style={{backgroundImage: `url(${imageUrl})`}}></div>
          <p className="crosshair__delete" onClick={() => removeCrosshair({ crosshair, id, name, team, imageUrl})}>Delete</p>
        </div>
      ))}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  savedCrosshairs: selectFavoriteCrosshairs,
  copied: selectIfCopied
});
const mapDispatchToProps = (dispatch) => ({
  removeCrosshair: settings => dispatch(removeCrosshair(settings)),
})


export default connect(mapStateToProps, mapDispatchToProps)(PlayersCrosshairs);
