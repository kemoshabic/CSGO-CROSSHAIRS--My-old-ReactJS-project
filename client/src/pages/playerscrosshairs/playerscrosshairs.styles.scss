.crosshair {
  display: flex;
  flex-direction: column;
  padding-top: 3rem;
  &__wrap {
    padding-bottom: 2rem;
    padding-left: 7rem;
    padding-top: 2rem;
    display: flex;
    align-items: center;
    border-bottom: 1px solid #fd074e;
    @media only screen and (max-width: 37.5em) {
      padding-bottom: 1rem;
      padding-left: 2rem;
      padding-top: 2rem;
    }
  }
  &__heading {
    color: #ffffff;
    font-size: 2.8rem;
    font-weight: 400;
    padding-right: 2rem;
    margin-bottom: -0.5rem;
    padding-left: 1.5rem;
  }
  &__team {
    color: #ffffff;
    font-size: 1.8rem;
    margin-bottom: -1.2rem;
    padding-right: 2rem;
  }
  
  &__img{
    height: 7rem;
    width: 9rem;
    background-color: transparent;
    background-repeat: no-repeat;
    background-position: center;
    @media only screen and (max-width: 37.5em) {
      background-size: cover;
      width: 5rem;
      margin-left: 2rem;
    }
  }
  &__delete {
    font-size: 2rem;
    color: red;
    text-decoration: underline;
    margin-bottom: -2.8rem;
    padding-left: 1.8rem;
    cursor: pointer;
    &:hover {
      background-image: linear-gradient(to right, #80ff72, #7ee8fa);
      background-clip: text;
      color: transparent;
    }
  }
}
