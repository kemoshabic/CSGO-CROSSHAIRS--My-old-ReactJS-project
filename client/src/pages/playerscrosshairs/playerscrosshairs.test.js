import React from "react";
import { shallow } from "enzyme";
import { PlayersCrosshairs } from "./playerscrosshairs.component";
describe("PlayersCrosshairs component", () => {
  let wrapper;
  let mockRemoveCrosshair;
  const mockSavedCrosshairs = [{
    crosshair: 'text',
    id: 1,
    name: 'text',
    team: 'text',
    imageUrl: 'text',
  }]
  beforeEach(() => {
    mockRemoveCrosshair = jest.fn();
    const mockProps = {
      savedCrosshairs: mockSavedCrosshairs,
      copied: false,
      removeCrosshair: mockRemoveCrosshair,
    };

    wrapper = shallow(<PlayersCrosshairs {...mockProps} />);
  });

  it("should render CheckoutPage component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call removeCrossair function when button is clicked", () => {
    wrapper.find(".crosshair__delete").simulate("click");
    expect(mockRemoveCrosshair).toHaveBeenCalled();
  });

  it("should not render crosshair preview when there is no data", () => {
    const mockProps = {
      savedCrosshairs: [],
      copied: false,
      removeCrosshair: mockRemoveCrosshair,
    };

    const newWrapper = shallow(<PlayersCrosshairs {...mockProps} />);
    expect(newWrapper.exists('.crosshair__wrap')).toBe(false);
  });

  it("should render only 1 saved crosshair and length should be 1", () => {
    expect(mockSavedCrosshairs).toHaveLength(1);
  });
});
