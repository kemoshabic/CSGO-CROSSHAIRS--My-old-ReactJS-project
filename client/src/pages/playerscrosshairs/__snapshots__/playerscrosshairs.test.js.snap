// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersCrosshairs component should render CheckoutPage component 1`] = `ShallowWrapper {}`;
